modules/memberships/infrastructure/stripe/operations/get-subscription-lifetime-collected-revenue.ts

```
@@ -0,0 +1,130 @@
+		amountPaidTotal:
+			params.accumulator.amountPaidTotal + params.invoice.amount_paid,
+		currency: params.accumulator.currency,
+	};
+}
+
+export function createGetSubscriptionLifetimeCollectedRevenueOperation(
+	dependencies: StripeOperationContext,
+): StripeGateway["getSubscriptionLifetimeCollectedRevenue"] {
+	const { stripe, logger, clock } = dependencies;
+
+	return async (subscriptionId) => {
```

Biome (2.4.9)

[error] Excessive complexity of 14 detected (max: 12).

(lint/complexity/noExcessiveCognitiveComplexity)